kml/serdes_gpx.cpp

```diff
@@ -101,20 +103,18 @@ bool GpxParser::Push(std::string_view tag)
   return true;
 }
 
+bool GpxParser::ValidCoordinatesPosition()
+{
+  return GetTagFromEnd(0) == gpx::kWpt || (GetTagFromEnd(0) == gpx::kTrkPt && GetTagFromEnd(1) == gpx::kTrkSeg) ||
+         (GetTagFromEnd(0) == gpx::kRtePt && GetTagFromEnd(1) == gpx::kRte);
+}
+
 void GpxParser::AddAttr(std::string const & attr, std::string const & value)
 {
   std::string attrInLowerCase = attr;
   strings::AsciiToLower(attrInLowerCase);
 
-  if (GetTagFromEnd(0) == gpx::kWpt)
-  {
-    if (attr == "lat")
-      m_lat = stod(value);
-    else if (attr == "lon")
-      m_lon = stod(value);
-  }
-  else if ((GetTagFromEnd(0) == gpx::kTrkPt && GetTagFromEnd(1) == gpx::kTrkSeg) ||
-           (GetTagFromEnd(0) == gpx::kRtePt && GetTagFromEnd(1) == gpx::kRte))
+  if (ValidCoordinatesPosition())
   {
     if (attr == "lat")
```

Comment:
coordinates are encoded equally in all 3 different cases in ValidCoordinatesPosition, right?

Reply:
Yes, lat/lon attributes for trkpt, wpt and rtept.